### PR TITLE
haproxy module wait

### PR DESCRIPTION
**Pull request by ggermis**
I was testing rolling updates with haproxy and found that the haproxy module simply does an enable/disable without checking if the service actually was successfully started / reachable. This caused all services to enter the DOWN state after running the ansible playbook.

I added a few options to the haproxy module to allow it to wait for a specific status before continuing and raising an error when it could not reach the expected state within a set number of retries

The options added were:
- wait (default = false): When state=enabled, wait until the service gets to the status 'UP'. When state=disabled, wait until the service gets to the status 'MAINT'
- wait_retries (default = 20): How many times to retry (show stat)
- wait_interval (default = 1): How many seconds to wait before the next retry

This ensures that our rolling updates never bring down all services. It will try to enable 'serial' number of apps at a time and if that block fails to reach the 'UP' status, will fail the ansible-playbook command, leaving the other services running and allowing us to intervene

### CODE REVIEW

**Comment by resmo** [comment]
LGTM
+1

**Comment by gregdek** [comment]
@ravibhure please review this patch -- if you give it +1, we will merge.

File: network/haproxy.py [hunk 1]
```diff
@@ -68,6 +68,21 @@
       - When disabling server, immediately terminate all the sessions attached to the specified server. This can be used to terminate long-running sessions after a server is put into maintenance mode, for instance.
     required: false
     default: false
+  wait:
```
Comment (bcoca) [inline network/haproxy.py:71]: you want to add version_added for new parameters for an existing module

Comment (ggermis) [inline network/haproxy.py:71]: added version_added: "2.0" to new parameters

**Comment by ravibhure** [comment]
+1 :+1:

**Comment by gregdek** [comment]
Merged. Thanks @ggermis for the PR, and thanks @resmo and @ravibhure for the reviews!

**Pull request merged**